kstreams/engine.py

```diff
@@ -144,6 +147,125 @@ async def send(
 
         return metadata
 
+    async def send_many(
+        self,
+        topic: str,
+        partition: int,
```

Comment:
Have you thought of calling this `default_partition` and then having `BatchEvent.partition`?
If you have a stream assigned to more than 1 partition, you could do:

```python
events = []
def stream(cr: ConsumerRecord):
    event = BatchEvent(partition=cr.partition)
    events.append(event) # then send them in batch
```